core/src/main/java/org/hisp/dhis/android/core/fileresource/internal/FileResourceUtil.kt

```diff
@@ -146,4 +149,38 @@ internal object FileResourceUtil {
     private fun logMessage(e: Exception) {
         e.message?.let { Log.v(FileResourceUtil::class.java.canonicalName, it) }
     }
+
+    internal fun computeImageDimension(orignalContentLength: Long?, maxContentLength: Long? = null): String {
+        if (orignalContentLength == null || maxContentLength == null) return DimensionSizeB.Medium.name
+
+        // Create list with dynamically set ORIGINAL size
+        val originalSize = DimensionSizeB.Original(orignalContentLength)
+        val sizes = listOf(
+            DimensionSizeB.Small,
+            DimensionSizeB.Medium,
+            originalSize,
+        )
+
+        // Filter out sizes requiring upscaling
+        val validSizes = sizes.filter { it.maxSizeB <= orignalContentLength }
+
+        // Step 3: Remove ORIGINAL if it exceeds MEDIUM max size
+        val limitedSizes = validSizes.filterNot {
+            it.name == originalSize.name && it.maxSizeB > DimensionSizeB.Medium.maxSizeB
+        }
+
+        // Select the largest size within maxContentLength
+        return limitedSizes.lastOrNull { it.maxSizeB <= maxContentLength }?.name ?: limitedSizes.first().name
```

Comment:
The `first()` method might return a `NoSuchElementException`when the list is empty. This will never happen because the application will only reach this part of the code when the contentLength is accepted, which implicitly implies that this list won't be empty. But it seems a bit fragile to me.

Another option could be to change the condition and use this method to determine if the `contentLength` is accepted or not. We could make a difference between FILE and IMAGE (which didn't exist before this PR):
- if it is a FILE, return ORIGINAL only if the contentLength is smaller than maxContentLength. Otherwise return null.
- if it is an IMAGE, return the corresponding dimension, having MEDIUM as the maximum. If there is no valid dimension, return null.

Then, this dimension could be used to determine if the contentLength is accepted: if the dimension is not null, it is accepted. Otherwise, it is not. From this point on, we will have a nonNull dimension and we could get rid of this.

Additionally, if the dimension is ORIGINAL we should skip the parameter `dimension` in the api request. We were always including this parameter (even for files) and it only makes sense for images.